### ISSUE TITLE
Sending a memo or message to recipient

### ISSUE THREAD

**Issue by afterconnery** [turtlecoin/turtlecoin-mobile-wallet#100]
Would it be possible to send a note or memo along with transaction. So that way the recipient would know what the received money is for. Like payment for "dinner" or "beard oil" etc. I don't know if this would also have to be available as an option for all the wallets that way I could send a memo from my proton wallet to your tonchan wallet.

**Comment by zpalmtree** [comment]
Is it possible? Yes
Do I want to add it? Not really
I really dislike the use of tx_extra for arbitary data, it's so messy and really needs standardizing otherwise we'll get people delimiting their custom fields in different ways, then different applications may end up reading random data created by another app as their custom data.
Furthermore, the current RPC call to fetch data doesn't even include tx_extra, since we don't need it and it's extra un-needed data which slows down syncing.
If we added a specific field for it in the transaction, or standardized tx_extra, I would be OK with adding it.

**Comment by 0des** [comment]
Perhaps things that are legitimate uses for tx_extra should get their own fields in the block template, and arbitrary data should be removed as a possibility until it can be implemented in a responsible way to minimize conflicts.

**Comment by zpalmtree** [comment]
I agree with that, separate fields are so much easier to handle and far simpler too. Then again, tx_extra does give you the option to implement pretty much anything... but I have yet to see it used for anything worthwhile

**Comment by zpalmtree** [comment]
If we did end up having a growing amount of custom fields, it would be worth considering adding params to /getwalletsyncdata specifying which stuff you want, to avoid the daemon doing extra processing and avoiding fetching data that is un-needed by the specific app.
